packages/telegram-stamper/README.md

```diff
@@ -0,0 +1,69 @@
+# @turnkey/telegram-stamper
+
+[![npm](https://img.shields.io/npm/v/@turnkey/telegram-stamper?color=%234C48FF)](https://www.npmjs.com/package/@turnkey/telegram-stamper)
+
+This package contains functions to store a turnkey api public/private key within [Telegram Cloud Storage](https://core.telegram.org/bots/webapps#cloudstorage). This package also handles stamping a Turnkey request with that api key. It is meant to be used with [`@turnkey/http`](https://www.npmjs.com/package/@turnkey/http)
+
+Preqrequisites: \
+Telegram Bot API >= 6.9
+
+Usage:
+
+Insert a new api key into Telegram Cloud Storage
+
+```ts
+import { TelegramStamper } from "@turnkey/telegram-stamper";
+import { TurnkeyClient } from "@turnkey/http";
+
+// create a new telegram stamper
+const stamper = new TelegramStamper({
+  apiPublicKey: "...",
+  apiPrivateKey: "...",
+});
+
+// initialize the telegram stamper, this stores the api key credentials in Telegram Cloud Storage for the first time
+try {
+  stamper.init();
+} catch (err) {
+  throw new Error(`Failed initializing Telegram Stamper: ${err}`)
+}
+
+// create a TurnkeyClient with the initialized Telegram Stamper
+const httpClient = new TurnkeyClient(
+  { baseUrl: "https://api.turnkey.com" },
+  stamper
+);
+
+// make a request with the client
+const whoamiResponse = await client.getWhoami({
+  organizationId: <ORGANIZATION_ID>,
+});
+```
+
+Use an exisiting key that has been previously stored in Telegram Cloud Storage
+
+```ts
+import { TelegramStamper } from "@turnkey/telegram-stamper";
+import { TurnkeyClient } from "@turnkey/http";
+
+// create a new telegram stamper
+const stamper = new TelegramStamper();
```

Comment:
I think the same can be said here, let's combine init and constructor?

```ts
// creates an API key, or gets the API key from cloudstorage default entry
const stamper = new TelegramStamper(); 

// creates an API key, or gets the API key from "MY_CUSTOM_ENTRY"
const stamper = new TelegramStamper({
    cloudStorageKey: "MY_CUSTOM_ENTRY"
});
```

Reply:
The init is actually an async method, so it needs to be separate from the constructor (i know there is no await in the readme so that was unclear, ive just added it locally)

Reply:
ah, I see. So that's a typescript limitation. Why does `const stamper = await new TelegramStamper(...)` not work 😭 

...did some quick Googling: https://stackoverflow.com/questions/35743426/async-constructor-functions-in-typescript...

And yeah, makes sense. The return type must be the type, not `Promise<Type>`. I like the alternative that's suggested though. We could have static method: `const stamper = await TelegramStamper.create(...);`

Reply:
This isn't really blocking btw; just food for thought for interfaces!

Reply:
Yeah, I used init as we have a similar `.init()` for the [iframestamper](https://github.com/tkhq/sdk/blob/main/packages/iframe-stamper/README.md) , so I wanted to have a similar flow between all of our sdk abstractions :)

Reply:
Yeah I hear you. Consistency vs. improvement of the interface. That's a tough one because I don't want to force us to rework all of the other stampers. But at the same time, there will never be a better opportunity to improve this interface. Once it's released it's much harder.

I'll leave this up to you and others to decide on this one; no strong feelings either way!

Reply:
I think the present is the best time to have change to move to best practices, I changed to a .create() method and made the constructor private. This was a decent sized refactor